### PR TITLE
Added additional compositions and identity

### LINKED ISSUE #211: Add additional composition aggregate types
There are use cases that warrant the following aggregate types.

* __incomplete_first_party_proprietary_only__ = Contains only first-party components or services, limited specifically to those that are proprietary.
* __incomplete_first_party_opensource_only__ = Contains only first-party components or services, limited specifically to those that are opensource.
*  __incomplete_third_party_proprietary_only__ = Contains only third-party components or services, limited specifically to those that are proprietary.
* __incomplete_third_party_opensource_only__ = Contains only third-party components or services, limited specifically to those that are opensource.

### PR DESCRIPTION
**Pull request by stevespringett**
Closes #211

### CODE REVIEW

File: tools/src/test/resources/1.5/valid-compositions-1.5.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@
                 <dependency ref="acme-application-1.0"/>
             </dependencies>
         </composition>
-        <composition>
+        <composition bom-ref="composition-2">
```
Comment (jkowalleck) [inline tools/src/test/resources/1.5/valid-compositions-1.5.xml:44]: please revert this line.
we need a showcase where no (optional) `bom-ref` attribute is set

File: tools/src/test/resources/1.5/valid-compositions-1.5.textproto [hunk 2]
```diff
@@ -39,11 +39,13 @@ dependencies {
   }
 }
 compositions {
+  bom_ref: "composition-1"
   aggregate: AGGREGATE_COMPLETE
   assemblies: "pkg:maven/partner/shaded-library@1.0"
   dependencies: "acme-application-1.0"
 }
 compositions {
+  bom_ref: "composition-1"
```
Comment (jkowalleck) [inline tools/src/test/resources/1.5/valid-compositions-1.5.textproto:48]: please revert this line.
we need a showcase where no (optional) `bom-ref` property is set

File: tools/src/test/resources/1.5/valid-compositions-1.5.json [hunk 3]
```diff
@@ -55,6 +56,7 @@
       ]
     },
     {
+      "bom-ref": "composition-2",
```
Comment (jkowalleck) [inline tools/src/test/resources/1.5/valid-compositions-1.5.json:59]: please revert this line.
we need a showcase where no (optional) `bom-ref` property is set

Comment (stevespringett) [reply tools/src/test/resources/1.5/valid-compositions-1.5.json]: Good catch. Updated test cases.

File: tools/src/test/resources/1.5/valid-compositions-1.5.textproto (hunk 2 above)
Comment (stevespringett) [reply tools/src/test/resources/1.5/valid-compositions-1.5.textproto]: Good catch. Updated test cases.

File: tools/src/test/resources/1.5/valid-compositions-1.5.xml (hunk 1 above)
Comment (stevespringett) [reply tools/src/test/resources/1.5/valid-compositions-1.5.xml]: Good catch. Updated test cases.

**Review by madpah — APPROVED** [review]

**Review by madpah — APPROVED** [review]

**Pull request merged**